modules/product/src/Form/AddToCartForm.php

```diff
@@ -166,4 +182,253 @@ function submitForm(array &$form, FormStateInterface $form_state) {
     ]));
   }
 
+  /**
+   * Builds the variation selection form.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   * @param \Drupal\commerce_product\Entity\ProductVariationInterface[] $variations
+   *   An array of product variations.
+   */
+  protected function variationForm(array &$form, FormStateInterface $form_state, array $variations) {
```

Comment:
The body of this method belongs in buildForm(), it's essential to the form structure so we gain nothing by splitting it out.